File: src/ddd/simple/service/teacher_module/TeacherService.java
package ddd.simple.service.teacher_module;

import ddd.base.persistence.EntitySet;
import ddd.simple.service.base.BaseServiceInterface;
import ddd.simple.entity.teacher_module.Teacher;

public interface TeacherService extends BaseServiceInterface {
	public Teacher saveTeacher(Teacher teacher) ;
	
	public int deleteTeacher(Long teacherId) ;
	
	public Teacher updateTeacher(Teacher teacher) ;
	
	public Teacher findTeacherById(Long teacherId) ;
	
	public Teacher findTeacherByOperator(Long operatorId);
	
	public EntitySet<Teacher> findAllTeacher() ;
 
}

File: WebContent/ddd/simple/view/teacher_module/js/teacher/TeacherService.js

define(["app"], function (app) {
	app.factory("TeacherService", function($resource) {
		
		var teacherService = $resource("../DD/Teacher/:action", {});
		teacherService.saveTeacher=function(teacher,sucesscb,errorcb)	{
			teacherService.save({action:"saveTeacher"},{teacher:teacher},sucesscb,errorcb);
		};
		
		teacherService.deleteTeacher=function(teacherId,sucesscb,errorcb)	{
			teacherService.save({action:"deleteTeacher"},{teacherId:teacherId},sucesscb,errorcb);
		};
		
		teacherService.updateTeacher=function(teacher,sucesscb,errorcb) {
			teacherService.save({action:"updateTeacher"},{teacher:teacher},sucesscb,errorcb);
		};
		 
		teacherService.findTeacherById=function(teacherId,sucesscb,errorcb) {
			teacherService.get({action:"findTeacherById",teacherId:teacherId},sucesscb,errorcb);
		};
		
		teacherService.findAllTeacher=function(sucesscb,errorcb) {
			teacherService.query({action:"findAllTeacher"},sucesscb,errorcb);
		}
		return teacherService;
})});
